src/classes/HiFiCommunicator.ts

```diff
@@ -159,13 +300,21 @@ export class HiFiCommunicator {
         if (onUsersDisconnected) {
             this.onUsersDisconnected = onUsersDisconnected;
         }
+        if (onConnectionStateChanged) {
+            this.onConnectionStateChanged = onConnectionStateChanged;
+        }
+        this._connectionRetryAndTimeoutConfig = HiFiConstants.DEFAULT_CONNECTION_RETRY_AND_TIMEOUT;
+        if (connectionRetryAndTimeoutConfig) {
+            Object.assign(this._connectionRetryAndTimeoutConfig, connectionRetryAndTimeoutConfig);
```

Comment:
This is a bug. It is side effecting the internal contents of the object at HiFiConstants.DEFAULT_CONNECTION_RETRY_AND_TIMEOUT, which will effect other HiFiCommunicator instances.

I think I would specify either HiFiConstants.DEFAULT_CONNECTION_RETRY_AND_TIMEOUT or the empty {} as the default value for the connectionRetryAndTimeoutConfig parameter of the constructor, and then replace all four lines here with:
```
this._connectionRetryAndTimeoutConfig = {}; // Or uses class field syntax if we're willing to require latest Safari.
Object.assign(this._connectionRetryAndTimeoutConfig, HiFiConstants.DEFAULT_CONNECTION_RETRY_AND_TIMEOUT, connectionRetryAndTimeoutConfig);
```

Reply:
Good eye -- thank you!!